src/shared/components/Snackbar/Snackbar.style.ts

```diff
@@ -1,72 +1,128 @@
-import { breakpoints, colors, sizes, transitions, typography, zIndex } from '@/shared/theme'
-import { css } from '@emotion/react'
 import styled from '@emotion/styled'
+import { colors, sizes, transitions, typography, zIndex } from '@/shared/theme'
+import { Button, Text } from '@/shared/components'
+import Icon from '../Icon'
+import { SnackbarVariant } from './Snackbar'
 
-const SNACKBAR_POSITION = 34
-const SNACKBAR_POSITION_MOBILE = 15
+type SnackbarWrapperProps = {
+  variant?: SnackbarVariant
+  snackbarHeight?: number
+}
 
-export const snackbarTransitions = css`
-  .snackbar-enter {
-    transform: translateX(-200%);
+type InnerWrapperProps = {
+  hasSubMessage?: boolean
+}
+
+export const SnackbarWrapper = styled.div<SnackbarWrapperProps>`
+  position: relative;
+  width: 100%;
+  height: 0;
+  background-color: ${({ variant }) => (variant === 'secondary' ? colors.gray[700] : colors.blue[500])};
+  z-index: ${zIndex.overlay};
+  overflow: hidden;
+  transform: translateY(500px) translateX(0);
+  &.snackbar-enter {
+    transform: translateY(500px) translateX(0);
+    height: 0;
+    margin-bottom: 0;
   }
-  .snackbar-enter-active {
-    transform: translateX(0);
+  &.snackbar-enter-active {
+    transform: translateY(0) translateX(0);
+    height: ${({ snackbarHeight }) => snackbarHeight && snackbarHeight}px;
+    margin-bottom: ${sizes(3)};
   }
-  .snackbar-exit {
-    transform: translateX(0);
+  &.snackbar-enter-done {
+    transform: translateY(0) translateX(0);
+    height: ${({ snackbarHeight }) => snackbarHeight && snackbarHeight}px;
+    margin-bottom: ${sizes(3)};
   }
-  .snackbar-exit-active {
-    transform: translateX(-200%);
+  &.snackbar-exit {
+    transform: translateY(0) translateX(0);
+    height: ${({ snackbarHeight }) => snackbarHeight && snackbarHeight}px;
+    margin-bottom: ${sizes(3)};
   }
-  .snackbar-enter-active {
-    transition: transform ${transitions.timings.loading} ${transitions.easing};
+  &.snackbar-exit-active {
+    transform: translateY(0) translateX(-150%);
+    height: 0;
+    margin-bottom: 0;
   }
-  .snackbar-exit-active {
-    transition: transform 800ms ${transitions.easing};
+
+  &.snackbar-enter-active {
+    transition: height ${transitions.timings.regular} ${transitions.easing},
+      margin-bottom ${transitions.timings.regular} ${transitions.easing},
+      transform ${transitions.timings.regular} ${transitions.easing} ${transitions.timings.regular};
+  }
+  &.snackbar-exit-active {
+    transition: transform ${transitions.timings.regular} ${transitions.easing},
+      height ${transitions.timings.regular} ${transitions.easing} ${transitions.timings.regular},
+      margin-bottom ${transitions.timings.regular} ${transitions.easing} ${transitions.timings.regular};
+  }
```

Comment:
```suggestion
  &.snackbar-enter-active,
  &.snackbar-exit-active {
    transition: height ${transitions.timings.regular} ${transitions.easing},
      margin-bottom ${transitions.timings.regular} ${transitions.easing},
      transform ${transitions.timings.regular} ${transitions.easing} ${transitions.timings.regular};
  }
```

Reply:
that cannot be really applied. These transitions are not the same. In `enter` there's delay on `transform` and in `exit` there's delay on `margin-bottom` and `height`

Reply:
My mistake, didn't notice